// Copyright 2020 Standard Cyborg
// 
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
// 
//     http://www.apache.org/licenses/LICENSE-2.0
// 
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.
syntax = "proto3";

import "standard_cyborg/proto/math/tensor.proto";


package standard_cyborg.proto.sc3d;

// An image with origin at the top-left corner
message Image {
  string frame = 1;
  
  math.Tensor pixels = 2;

  // ColorSpace enumerates all allowable image pixel color space representations.
  // https://en.wikipedia.org/wiki/Color_space
  enum ColorSpace {
    COLOR_SPACE_UNKNOWN = 0;
  
    // A linear value as would be used to represent grayscale or depth.
    COLOR_SPACE_LINEAR = 1;
    
    // https://en.wikipedia.org/wiki/SRGB
    COLOR_SPACE_SRGB = 2;

    // This image is actually 1-channel depth image
    COLOR_SPACE_DEPTH = 100;
  }

  // How to interpret `pixels`
  ColorSpace color_space = 10;
}
